### DISCUSSION TITLE
Add an easy way to see history of edits for a field

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5660] [Feature Requests]
https://github.com/specify/specify7/issues/1703 was great, but we need something similar for a single field (you can modify the query, but that's too complicated)

Maybe alt+click on label will open a dialog with edit history for that field

**Comment by CarolineDenis** [comment]
@maxpatiiuk could we meet to discuss this?

**Comment by maxpatiiuk** [comment]
Sure,
If you go into the Form Meta menu, there would be an "Edit History" button - but it shows history of edits of all fields for this resource. You can of course modify the query manually to filter results down to a single field, but that isn't very convinient.

The task here is to make it easier to see field-specific edit history. My suggestion was to have an alt+click option that will open a dialog similar to "Edit History" but scoped to history for this field only.

Outline:
- Figure out if seeing field-specific edit history is valuable if implemented using alt+click OR if we need to consider alternative UI for implementing this (i.e, maybe adding support for "group by" in query builder would provide a better solution)
- If this is only marginally more useful than seeing all fields at once (i.e, perhaps because several fields could be changed at the same time, and you can only see that if looking at the entire edit history), then it might not be worth it
  - Potentially, a more creative way to display the history would be useful:
    - something like in google docs would be really cool: you see a list of dates when changes were made, and can click on any date to see the form, and the form will highlight what things were changed on that day
    - of course, there are several difficulties here, i.e:
       - If you are looking at a Collection Object, and it has a Collection, and Collector had it's last name changed, even though that wasn't editing the Collection Object table, perhaps it would be useful to see that kind of change?
       - If Collection Object had a collector, then it was changed to a different collector and the first one was deleted, how should the form display a now-deleted resource?

**Comment by maxpatiiuk** [comment]
In summary, while we would like to make it easy to see an edit history for a field, implementing such feature and making it useful might not be easy 😄
